### PR TITLE
Schema update for queries with lists of transactions

### PR DESCRIPTION
**Pull request by MBoldyrev**
### Description of the Change

[IR-20](https://jira.hyperledger.org/projects/IR/issues/IR-20)
This change adds the pagination metadata to some queries that request transaction lists: GetAccountTransactions and GetAccountAssetTransactions. It does not concern GetPendingTransactions as it is expected to be rewritten soon.

### Benefits

A small step towards great feature.

### Possible Drawbacks 

bugs.

### Usage Examples or Tests

When a client requests some sort of transaction list, the request must include pagination metadata. That one _must_ have the page size set  to the maximum amount of transactions the client wants to get at once, and _may_ have the first transaction hash field set to the hash of the transaction, from which the returned list is supposed to start.

### CODE REVIEW

File: shared_model/interfaces/common_objects/types.hpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ namespace shared_model {
       using AccountDetailKeyType = std::string;
       /// Type of account detail value
       using AccountDetailValueType = std::string;
-      /// Type of a number of transactions in block
+      /// Type of a number of transactions in block and query responce page
```
Comment (Akvinikym) [inline shared_model/interfaces/common_objects/types.hpp]: query responSe page

File: shared_model/backend/protobuf/queries/impl/proto_get_account_asset_transactions.cpp [hunk 2]
```diff
@@ -39,5 +41,11 @@ namespace shared_model {
       return account_asset_transactions_.asset_id();
     }
 
+    std::unique_ptr<interface::TxPaginationMeta>
```
Comment (Akvinikym) [inline shared_model/backend/protobuf/queries/impl/proto_get_account_asset_transactions.cpp]: Why do we want to return a unique_ptr? Is the meta going to be ever changed? If not, why not returning a const ref, as it is done in other methods of the class?

File: shared_model/backend/protobuf/queries/impl/proto_tx_pagination_meta.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+namespace types = shared_model::interface::types;
+
+using namespace shared_model::proto;
+
+TxPaginationMeta::TxPaginationMeta(const TransportType &query)
+    : CopyableProto(query) {}
+
+TxPaginationMeta::TxPaginationMeta(TransportType &&query)
+    : CopyableProto(
+          static_cast<std::remove_reference<TransportType>::type &&>(query)) {}
```
Comment (Akvinikym) [inline shared_model/backend/protobuf/queries/impl/proto_tx_pagination_meta.cpp]: Why such complex structure and not a simple `move`?

File: shared_model/interfaces/queries/tx_pagination_meta.hpp [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      virtual types::TransactionsNumberType pageSize() const = 0;
+
+      /// Get the first requested transaction hash, if provided.
+      virtual boost::optional<types::HashType> firstTxHash() const = 0;
+
+      std::string toString() const override;
+
+      bool operator==(const ModelType &rhs) const override;
+    };
+  }  // namespace interface
```
Comment (Akvinikym) [inline shared_model/interfaces/queries/tx_pagination_meta.hpp]: Empty line should be inserted

File: shared_model/backend/protobuf/queries/impl/proto_get_account_asset_transactions.cpp (hunk 2 above)
Comment (MBoldyrev) [reply shared_model/backend/protobuf/queries/impl/proto_get_account_asset_transactions.cpp]: The other methods return primitive objects, while this requires construction. Returning a constref would require us to keep a copy of `proto::TxPaginationMeta` inside all proto query classes. I think a unique pointer is a more clear way to provide `proto::TxPaginationMeta` when it is needed.

Comment (Akvinikym) [reply shared_model/backend/protobuf/queries/impl/proto_get_account_asset_transactions.cpp]: Please, consider this issue again before merging

File: shared_model/backend/protobuf/queries/proto_tx_pagination_meta.hpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SHARED_PROTO_MODEL_QUERY_TX_PAGINATION_META_HPP
+#define IROHA_SHARED_PROTO_MODEL_QUERY_TX_PAGINATION_META_HPP
```
Comment (Akvinikym) [inline shared_model/backend/protobuf/queries/proto_tx_pagination_meta.hpp]: Include guards in other files in this folder are called `IROHA_SHARED_MODEL_PROTO_...`. Please, rename this one according to the common scheme

File: shared_model/interfaces/queries/tx_pagination_meta.hpp [hunk 6]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SHARED_INTERFACE_MODEL_QUERY_TX_PAGINATION_META_HPP
+#define IROHA_SHARED_INTERFACE_MODEL_QUERY_TX_PAGINATION_META_HPP
```
Comment (Akvinikym) [inline shared_model/interfaces/queries/tx_pagination_meta.hpp]: The same

File: shared_model/backend/protobuf/queries/impl/proto_get_account_asset_transactions.cpp (hunk 2 above)
Comment (kamilsa) [reply shared_model/backend/protobuf/queries/impl/proto_get_account_asset_transactions.cpp]: Here we would need to create an object every time pagination meta object is invoked, whereas it is only access methods. Please change to cont ref as it is done for example for amount in transfer asset. It is okay to construct  pagination meta

**Pull request merged**